File: Python/String-1/con_cat.py

# given two strings, append them together (known as "concatenation") and return the result
# however, if the concatenation creates a double-char, then omit one of the chars
def con_cat(a, b):
    s = ''

    if a and not b:
        s = a
    if not a and b:
        s = b
    if a and b:
        if a[-1:] == b[0]:
            s = a[:-1] + b
        else:
            s = a + b

    return s

print(con_cat('abc', 'cat'))
print(con_cat('dog', 'cat'))
print(con_cat('abc', ''))

File: JavaScript/String-1/conCat.js

// given two strings, append them together (known as 'concatenation') and return the result
// however, if the concatenation creates a double-char, then omit one of the chars, so 'abc' and 'cat' yields 'abcat'
const conCat = (a, b) => {
    let str = '';

    if (a.length > 0 && b.length > 0 && a[a.length - 1] === b[0]) {
        str = a + b.substring(1);
    } else if (a.length > 0 && b.length === 0) {
        str = a;
    } else if (a.length === 0 && b.length > 0) {
        str = b;
    } else {
        str = a + b;
    }

    return str;
};

console.log(conCat('abc', 'cat'));
console.log(conCat('dog', 'cat'));
console.log(conCat('abc', ''));
